### PR TITLE
Update introduction_to_the_backend_lesson.md

### PR DESCRIPTION
**Pull request by NerdsAreIn**
I found this lesson quite lacking, in terms of both content and grammar, and I had to do quite a bit of Googling to gain even a basic understanding of what the back end actually is.

I deleted the second resource in the Assignment, as I would argue that it's more a promo piece about getting into back-end development/being a back-end developer than the back end itself. I added two resources - the best I could find on the subject, after viewing quite a lot of search results.

I added basic definitions of the front end and back end, which I feel are required, and made a number of grammatical revisions.

Finally, I don't really understand the second learning outcome question. Perhaps it's the wording. What reason IS there to install back-end languages on one's computer other than to be able to run them...? The reasons do not appear to be "listed above"...

 - [x] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).

#### 1.Describe the changes made:

...your text here

#### 2. If this PR is related to an open issue please reference it with the `#` operator and the issue number below:

#XXXXX

### CODE REVIEW

File: foundations/the_back_end/introduction_to_the_backend_lesson.md [hunk 1]
```diff
@@ -27,6 +30,6 @@ As we covered back in the [Installations section](https://www.theodinproject.com
 This section contains questions for you to check your understanding of this lesson. If you're having trouble answering the questions below on your own, review the material above to find the answer.
 
 * What is back-end development?
-* How many parts does the back-end consist of?
-* Name each part of the back-end.
-* Where does the back-end store its data?
+* How many parts does the back end consist of?
```
Comment (twalton83) [inline foundations/the_back_end/introduction_to_the_backend_lesson.md:33]: In the learning outcomes, we use the hyphen, so it's probably best to maintain that here.

File: foundations/the_back_end/introduction_to_the_backend_lesson.md [hunk 2]
```diff
@@ -1,24 +1,27 @@
...
+* Why are we required to install back-end languages on our computer (besides to run them)?
 
 ### Assignment
 
 <div class="lesson-content__panel" markdown="1">
 
   1. Check out [this blog post on back end vs front end programming](http://blog.teamtreehouse.com/i-dont-speak-your-language-frontend-vs-backend) for a quick refresher on the difference between the two.
-  2. Read over [this quick interview with Matt Jording](https://www.bizjournals.com/washington/inno/stories/news/2014/03/31/what-is-back-end-web-development.html) about what back end web development is.
+  2. [A good and short overview of the back end] (https://techterms.com/definition/backend).
+  3. A [much more detailed and thorough explanation] of the various aspects of the back end is provided by Codecademy, including a step-by-step breakdown of what happens in the back end when a browser makes a request to a server.
```
Comment (twalton83) [inline foundations/the_back_end/introduction_to_the_backend_lesson.md:24]: Looks like you missed adding the link here.

**Review by twalton83 — COMMENTED** [review]
Nothing sticks out to me aside from a few small things, let's see if other maintainers have any opinions about these changes.

File: foundations/the_back_end/introduction_to_the_backend_lesson.md (hunk 1 above)
Comment (NerdsAreIn) [reply foundations/the_back_end/introduction_to_the_backend_lesson.md:33]: But "the back end" is a noun phrase, whereas "back-end", as found elsewhere, is a compound adjective.

File: foundations/the_back_end/introduction_to_the_backend_lesson.md [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
+* What is the difference between front-end and back-end development?
+* Why are we required to install back-end languages on our computer (besides to run them)?
 
 ### Assignment
 
 <div class="lesson-content__panel" markdown="1">
 
   1. Check out [this blog post on back end vs front end programming](http://blog.teamtreehouse.com/i-dont-speak-your-language-frontend-vs-backend) for a quick refresher on the difference between the two.
-  2. Read over [this quick interview with Matt Jording](https://www.bizjournals.com/washington/inno/stories/news/2014/03/31/what-is-back-end-web-development.html) about what back end web development is.
+  2. [A good and short overview of the back end] (https://techterms.com/definition/backend).
```
Comment (marvingay) [inline foundations/the_back_end/introduction_to_the_backend_lesson.md:23]: The links present here are not properly formatted (see: the previous change / other links) no space between the brackets and parenthesis.

File: foundations/the_back_end/introduction_to_the_backend_lesson.md (hunk 1 above)
Comment (marvingay) [reply foundations/the_back_end/introduction_to_the_backend_lesson.md:33]: I'm a proponent of either leaving it hyphenated or changing it to "backend" in all places.

I think colloquialism in the context of this particular domain should trump what is considered to be "proper".

**Review by marvingay — CHANGES_REQUESTED** [review]

File: foundations/the_back_end/introduction_to_the_backend_lesson.md (hunk 3 above)
Comment (NerdsAreIn) [reply foundations/the_back_end/introduction_to_the_backend_lesson.md]: Hi. I made the requested edit. Hope I did it right.

File: foundations/the_back_end/introduction_to_the_backend_lesson.md (hunk 1 above)
Comment (NerdsAreIn) [reply foundations/the_back_end/introduction_to_the_backend_lesson.md:33]: Ok, but I don't think "back end" in any form is colloquial...? In another lesson, there's a clause that says "patch 'er up", which I would say is colloquial.

I think it's about consistency above all, and also clarity, and I think that a distinction between the noun and adjective enhances that.

**Comment by NerdsAreIn** [comment]
What happens now...?

**Comment by marvingay** [comment]
@NerdsAreIn All instances of front-end and back-end aren't hyphenated. Unless the files I'm being shown are out of sync.

**Comment by ZachBaird** [comment]
I think we should leave "back-end"  is used as an adjective to something like "languages". Ex. "back-end languages." This is, ofc, good grammar.

In the cases where we just refer to the "back end", I really do think it should just be "backend". There really isn't an industry standard here when we write "back end", "back-end", or "backend". Due to this, and how the backend is discussed as a cohesive idea amongst English-speaking developers (plus it's less characters to type), it is my preference.

This entirely ignores whatever standard we follow across the site; assuming we have a standard. TOP content is written by a wide array of contributors. But maybe this is a start to having some standard language and terminology on stuff.

**Comment by NerdsAreIn** [comment]
The proposals of @ZachBaird seem fine to me, though I do prefer "back end". Of course, all instances of "front-end" and "frontend" should likewise be changed to match - preferably course-wide.

**Comment by ZachBaird** [comment]
> Of course, all instances of "front-end" and "frontend" should likewise be changed to match - preferably course-wide

Yeah, I agree. Consistency is important. Feel free to put up a commit for those changes since we seem to be coming to an agreement.

Thanks for helping with the writing :)

**Comment by twalton83** [comment]
I also agree with @ZachBaird, and this is a good opportunity to begin to standardize.

**Comment by I3uckwheat** [comment]
> The proposals of @ZachBaird seem fine to me, though I do prefer "back end". Of course, all instances of "front-end" and "frontend" should likewise be changed to match - preferably course-wide.

I really think we should stick to "backend" and "frontend". It makes it easy to search for and removes ambiguity and relies less on the context of the term. There is no real standard for this. 

As I write this, I realize I am just re-iterating what @ZachBaird said. I agree with him.

**Review by marvingay — APPROVED** [review]

**Review by marvingay — APPROVED** [review]

**Pull request merged**